datahub-web-react/src/app/entity/chart/profile/ChartProfile.tsx

```diff
@@ -23,6 +24,20 @@ const ENABLED_TAB_TYPES = [TabType.Ownership, TabType.Sources];
 
 export default function ChartProfile({ urn }: { urn: string }) {
     const { loading, error, data } = useGetChartQuery({ variables: { urn } });
+    const [updateChart] = useUpdateChartMutation({
+        update(cache, { data: newChart }) {
+            cache.modify({
+                fields: {
+                    chart() {
+                        cache.writeQuery({
```

Comment:
Curious - what do lines 28 through 36 accomplish?

Reply:
Great question- this updates the apollo cache with new data when the mutation response comes back. This way, we don't need to refetch chart data or keep track of the partial update in local state